sdk/eventhub/Azure.Messaging.EventHubs/src/EventHubsConnectionStringProperties.cs

```diff
@@ -0,0 +1,347 @@
+            {
+                builder
+                    .Append(SharedAccessKeyNameToken)
+                    .Append(TokenValueSeparator)
+                    .Append(SharedAccessKeyName)
+                    .Append(TokenValuePairDelimiter)
+                    .Append(SharedAccessKeyValueToken)
+                    .Append(TokenValueSeparator)
+                    .Append(SharedAccessKey)
+                    .Append(TokenValuePairDelimiter);
+            }
+
+            return builder.ToString();
+        }
+
+        /// <summary>
+        ///   Performs the actions needed to validate the set of connection string properties for connecting to the
+        ///   Event Hubs service.
+        /// </summary>
+        ///
+        /// <param name="explicitEventHubName">The name of the Event Hub that was explicitly passed independent of the connection string, allowing easier use of a namespace-level connection string.</param>
+        /// <param name="connectionStringArgumentName">The name of the argument associated with the connection string; to be used when raising <see cref="ArgumentException" /> variants.</param>
+        ///
+        /// <exception cref="ArgumentException">In the case that the properties violate an invariant or otherwise represent a combination that is not permissible, an appropriate exception will be thrown.</exception>
+        ///
+        internal void Validate(string explicitEventHubName,
+                               string connectionStringArgumentName)
+        {
+            // The Event Hub name may only be specified in one of the possible forms, either as part of the
+            // connection string or as a stand-alone parameter, but not both.  If specified in both to the same
+            // value, then do not consider this a failure.
+
+            if ((!string.IsNullOrEmpty(explicitEventHubName))
+                && (!string.IsNullOrEmpty(EventHubName))
+                && (!string.Equals(explicitEventHubName, EventHubName, StringComparison.InvariantCultureIgnoreCase)))
+            {
+                throw new ArgumentException(Resources.OnlyOneEventHubNameMayBeSpecified, connectionStringArgumentName);
+            }
+
+            // The connection string may contain a precomputed shared access signature OR a shared key name and value,
+            // but not both.
+
+            if ((!string.IsNullOrEmpty(SharedAccessSignature))
+                && ((!string.IsNullOrEmpty(SharedAccessKeyName)) || (!string.IsNullOrEmpty(SharedAccessKey))))
+            {
+                throw new ArgumentException(Resources.OnlyOneSharedAccessAuthorizationMayBeSpecified, connectionStringArgumentName);
+            }
+
+            // Ensure that each of the needed components are present for connecting.
+
+            var hasSharedKey = ((!string.IsNullOrEmpty(SharedAccessKeyName)) && (!string.IsNullOrEmpty(SharedAccessKey)));
+            var hasSharedSignature = (!string.IsNullOrEmpty(SharedAccessSignature));
```

Comment:
Not.... quite.   The above says:  "if I have a signature and I have either a key or value, then my intent was not clear because I specified all or parts of both."  Using `hasSharedKey` would change the semantics to "If I have a signature and I have BOTH a key and value..."    

The reason that I used this form was to ensure that we don't assume that having a SAS alone is a clear statement of intent.   If I give you a SAS and some part of a key/value pair, then I'm giving you two forms of authorization, though one is not in a valid form.